src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityDiode.java

```diff
@@ -23,16 +26,20 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumHand;
 import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.world.World;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.Arrays;
 import java.util.List;
 
 import static gregtech.api.capability.GregtechDataCodes.AMP_INDEX;
 
-public class MetaTileEntityDiode extends TieredMetaTileEntity {
+public class MetaTileEntityDiode extends MetaTileEntityMultiblockPart implements EnergyContainerHandler.IEnergyChangeListener {
```

Comment:
Why was this changed to implement this Energy Change listener when it does not do anything with it?

In fact, this interface seems unused entirely, and from commit history was last used to update comparator values for machines, which was removed in 2019 in CE.